zaza/openstack/utilities/openstack.py

```diff
@@ -483,8 +484,68 @@ def get_admin_net(neutron_client):
             return net
 
 
+def _plumb_guest_nic(server_name, mac_address, dvr_mode=None):
+    """In guest server_name, add port with mac_address to netplan.
+
+    :param server_name: Hostname of instance
+    :type server_name: string
+    :param mac_address: mac address of nic to be added to netplan
+    :type mac_address: string
+    :param dvr_mode: Using DVR mode or not
+    :type dvr_mode: boolean
+    """
+    if dvr_mode:
+        application_name = 'neutron-openvswitch'
+    else:
+        application_name = 'neutron-gateway'
+
+    unit_name = juju_utils.get_unit_name_from_host_name(
+        server_name, application_name)
+
+    run_cmd_nic = "ip addr|grep {} -B1".format(mac_address)
+    interface = model.run_on_unit(unit_name, run_cmd_nic)
+    interface = interface['Stdout'].split(' ')[1].replace(':', '')
```

Comment:
I think you can simplify this a lot by using ```ip -f link -br -o addr |grep {}``` It will only return one line and the interface name won't have a ':' appended.